### PR TITLE
AdopterAnalysis: diagnose several issues within .swiftinterface

### PR DESCRIPTION
**Pull request by nkcsgexi**
1. missing `-enable-library-evolution`
2. missing a library level flag
3. wrong library level flag

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please test

File: Sources/SwiftDriver/Jobs/PrebuiltModulesJob.swift [hunk 1]
```diff
@@ -250,6 +276,12 @@ public struct PrebuiltModuleInput {
   }
 }
 
+public enum AdopterIssueKind: String, Codable {
+  case LibraryEvolutionDisabled
+  case LibraryLevelMissing
+  case LibraryLevelWrong
```
Comment (cltnschlosser) [inline Sources/SwiftDriver/Jobs/PrebuiltModulesJob.swift:282]: Should these be lowercased? With custom uppercase raw values if that’s needed.

File: Sources/SwiftDriver/Jobs/PrebuiltModulesJob.swift [hunk 2]
```diff
@@ -271,6 +304,27 @@ public class SwiftAdopter: Codable {
...
+      issues.append(.LibraryLevelMissing)
+    }
+    if libLevel == .spi && !isPrivate {
+      issues.append(.LibraryLevelWrong)
+    }
+    if libLevel == .api && isPrivate {
+      issues.append(.LibraryLevelWrong)
+    }
+    if !flags.contains("-enable-library-evolution") {
+      issues.append(.LibraryEvolutionDisabled)
```
Comment (cltnschlosser) [inline Sources/SwiftDriver/Jobs/PrebuiltModulesJob.swift:325]: Is it always the case that library evolution should be enabled for this? Not super familiar with this area.

File: Sources/SwiftDriver/Jobs/PrebuiltModulesJob.swift (hunk 1 above)
Comment (nkcsgexi) [reply Sources/SwiftDriver/Jobs/PrebuiltModulesJob.swift]: We totally should. Fixed.

File: Sources/SwiftDriver/Jobs/PrebuiltModulesJob.swift (hunk 2 above)
Comment (nkcsgexi) [reply Sources/SwiftDriver/Jobs/PrebuiltModulesJob.swift]: Yeah, emitting textual interface and enabling library-evolution are tied together. Using one without the other should raise an alarm.

**Comment by nkcsgexi** [comment]
@swift-ci please test

**Review by cltnschlosser — APPROVED** [review]

**Pull request merged**